server/master/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -1427,9 +1406,12 @@ public void run() {
 
     final long deadline = System.currentTimeMillis() + MAX_CLEANUP_WAIT_TIME;
     statusThread.join(remaining(deadline));
-    replicationWorkAssigner.join(remaining(deadline));
-    replicationWorkDriver.join(remaining(deadline));
-    replAddress.server.stop();
+    if (replicationWorkAssigner != null)
+      replicationWorkAssigner.join(remaining(deadline));
+    if (replicationWorkDriver != null)
+      replicationWorkDriver.join(remaining(deadline));
+    TServerUtils.stopTServer(replServer.get());
```

Comment:
Is it ok to pass stopTServer null?

Reply:
could do 

```java
if(replServer.get() != null){
     replicationWorkAssigner.join(remaining(deadline));		￼   
￼     replicationWorkDriver.join(remaining(deadline));		￼ 
￼     replAddress.server.stop();
}

```

Reply:
Yeah the TServerUtils method checks for null initially. I had something similar earlier but Josh suggest to always stop the thread.